www/src/pages/docs/50_StoreMapping.md

```diff
@@ -0,0 +1,430 @@
+        val nameStore = customerStore.map(Person.name())
+        input {
+            value(nameStore.data)
+            changes.values() handledBy nameStore.update
+        }
+    }
+    else {
+        p { + "no customer selected" }
+    }
+}
+```
+
+### Handling nullable States in `Store`s
+
+If you have a `Store` with a nullable state, you can use `mapNull` to derive a non-nullable `Store` from it,
+that transparently translates a `null`-value from its parent `Store` to the given default-value and vice versa.
+
+In the following case, when you enter some text in the input and remove it again,
+you will have a state of `null` in your `nameStore`:
+
+```kotlin
+val nameStore = storeOf<String?>(null)
+
+render {
+    input {
+        nameStore.mapNull("").also { formStore ->
+            value(formStore.data)
+            changes.values() handledBy formStore.update
+        }
+    }
+}
+```
+
+In real world, you will often come across nullable attributes of complex entities. Then you can call `mapNull`
+directly on the `Store` you create to use with your form elements:
+
+```kotlin
+@Lenses
+data class Person(val name: String?)
+
+//...
+
+val personStore = storeOf(Person(null))
+
+//...
+
+val nameStore = personStore.map(Person.name()).mapNull("")
+```
+
+### Formatting Values
+
+In html you can only use `String`s in your attributes like in the `value` attribute of `input {}`. To use other data
+types in your model you have to specify how to represent a specific value as `String` (e.g. Number, Currency, Date).
+When you work with `input {}` you also need parse the entered text back to your data type.
+For all Kotlin basic types there is a convenience function `asString()` which generates a `Lens` from this type to 
+`String` and vice versa. Therefore, it calls internally the `T.toString()` and `String.toT()` functions.
+
+```kotlin
+@Lenses
+data class Person(val age: Int)
+
+val ageLens: Lens<Person, Int> = Person.age() // cannot be used in tag attributes
+val ageLensAsString: Lens<Person, String> = Person.age().asString() // now it is usable
+```
+
+fritz2 also provides a special `lensOf()` function for creating a `Lens<P, String>` for special types that are not 
+basic:
+
+```kotlin
+fun <P> lensOf(format: (P) -> String, parse: (String) -> P): Lens<P, String>
+```
+
+If you use other types like `kotlinx.datetime.LocalDate` in your data classes, you have to specify a special lens for it.
+This lens then converts the value to a `String` and vice versa.
+```kotlin
+import kotlinx.datetime.*
+
+@Lenses
+data class Person(val birthday: LocalDate)
+
+object Formats {
+    val date: Lens<LocalDate, String> = lensOf(LocalDate::toString, String::toLocalDate)
+}
+```
+
+Now you can use the `Formats.date` lens for deriving appropriate stores:
+
+```kotlin
+val personStore = storeOf<Person>(Person(LocalDate(1990, 1, 1)))
+
+val birthday: Store<String> = personStore.map(Person.birthday() + Formats.date)
```

Comment:
Is there anywhere mentioned, that you can combine two lenses with the plus operator in general?

Reply:
No!